libraries/rush-lib/src/cli/actions/DeployAction.ts

```diff
@@ -150,7 +150,11 @@ export class DeployAction extends BaseRushAction {
         this.rushConfiguration.defaultSubspace
       );
       transformPackageJson = pnpmfileConfiguration.transform.bind(pnpmfileConfiguration);
-      if (!scenarioConfiguration.json.omitPnpmWorkaroundLinks) {
+      if (this.rushConfiguration.subspacesFeatureEnabled) {
+        const rushConfigurationProject = this.rushConfiguration.getProjectByName(mainProjectName);
+        const subspace = rushConfigurationProject?.subspace;
+        if (subspace) pnpmInstallFolder = subspace.getSubspaceTempFolder();
+      } else if (!scenarioConfiguration.json.omitPnpmWorkaroundLinks) {
```

Comment:
@jzhang026 I'm unusure about the logic here.  If I'm reading the code correctly:

- **Prior to this PR:** If `omitPnpmWorkaroundLinks=true`, then `pnpmInstallFolder=commonTempFolder` otherwise `pnpmInstallFolder=undefined` 
- **After this PR:** If `subspacesFeatureEnabled=true`, then `pnpmInstallFolder=getSubspaceTempFolder()` unless `mainProjectName` cannot be found in **rush.json** for some reason (?)

This seems inconsistent. @william2958's plan was that Rush 6 will retire the `subspacesFeatureEnabled` setting, basically treating it as always enabled.  (In other words, a classic single-lockfile monorepo will then be represented as using the "default" subspace only, rather than as a different usage mode.)  Therefore we expect the behavior to be consistent.